autogen/agentchat/groupchat.py

```diff
@@ -177,3 +182,48 @@ def run_chat(
             speaker.send(reply, self, request_reply=False)
             message = self.last_message(speaker)
         return True, None
+
+    async def a_run_chat(
```

Comment:
Lot of code duplication in run chat vs. run async chat. Please rewrite the function so common code parts are in a different local function.

Also change the name of the function to async_run_chat

Reply:
@pcdeadeasy I think the rest of the (existing) code is using a_ prefix for async functions. See

https://github.com/microsoft/autogen/blob/80954e4b8d0752fd4772f339dee419fbf1debc6f/autogen/agentchat/conversable_agent.py#L533